scipy/sparse/linalg/_eigen/lobpcg/tests/test_lobpcg.py

```diff
@@ -52,53 +57,24 @@ def ElasticRod(n):
     return A, B
 
 
-def MikotaPair(n):
-    """Build a pair of full diagonal matrices for the generalized eigenvalue
-    problem. The Mikota pair acts as a nice test since the eigenvalues are the
-    squares of the integers n, n=1,2,...
-    """
-    x = np.arange(1, n+1)
-    B = diag(1./x)
-    y = np.arange(n-1, 0, -1)
-    z = np.arange(2*n-1, 0, -2)
-    A = diag(z)-diag(y, -1)-diag(y, 1)
-    return A, B
-
-
-def compare_solutions(A, B, m):
-    """Check eig vs. lobpcg consistency.
+@pytest.mark.filterwarnings("ignore:The problem size")
+@pytest.mark.parametrize("n", [10, 20])
+@pytest.mark.filterwarnings("ignore:Exited at iteration")
+@pytest.mark.filterwarnings("ignore:Exited postprocessing")
+def test_ElasticRod(n):
+    """Check eigh vs. lobpcg consistency for elastic rod model.
     """
-    n = A.shape[0]
+    A, B = ElasticRod(n)
+    m = 2
     rnd = np.random.RandomState(0)
-    V = rnd.random((n, m))
-    X = orth(V)
+    X = rnd.standard_normal((n, m))
     eigvals, _ = lobpcg(A, X, B=B, tol=1e-2, maxiter=50, largest=False)
     eigvals.sort()
-    w, _ = eig(A, b=B)
+    w, _ = eigh(A, b=B)
     w.sort()
     assert_almost_equal(w[:int(m/2)], eigvals[:int(m/2)], decimal=2)
 
 
-def test_Small():
-    A, B = ElasticRod(10)
-    with pytest.warns(UserWarning, match="The problem size"):
-        compare_solutions(A, B, 10)
-    A, B = MikotaPair(10)
-    with pytest.warns(UserWarning, match="The problem size"):
-        compare_solutions(A, B, 10)
-
-
-def test_ElasticRod():
-    A, B = ElasticRod(20)
-    with pytest.warns(UserWarning, match="Exited at iteration"):
-        compare_solutions(A, B, 2)
-
-
-def test_MikotaPair():
```

Comment:
Other than switching to the new MikotaPair why are the changes here needed?

Reply:
I forgot already. I guess mostly just general cleaning and unification/beatification.